nengo/builder/connection.py

```diff
@@ -6,19 +6,30 @@
 from nengo.builder.builder import Builder
 from nengo.builder.ensemble import gen_eval_points, get_activities
 from nengo.builder.node import SimPyFunc
-from nengo.builder.operator import DotInc, ElementwiseInc, PreserveValue, Reset
+from nengo.builder.operator import (
+    DotInc, ElementwiseInc, PreserveValue, Reset, SlicedCopy)
 from nengo.builder.signal import Signal
 from nengo.builder.synapses import filtered_signal
 from nengo.connection import Connection
 from nengo.ensemble import Ensemble, Neurons
 from nengo.neurons import Direct
 from nengo.node import Node
-from nengo.utils.builder import full_transform
 from nengo.utils.compat import is_iterable, itervalues
 
 
 BuiltConnection = collections.namedtuple(
-    'BuiltConnection', ['decoders', 'eval_points', 'transform', 'solver_info'])
+    'BuiltConnection', ['eval_points', 'solver_info', 'weights'])
+
+
+def get_sliced_signal(signal, s, model):
+    assert signal.ndim == 1
```

Comment:
- Rename to `sliced_signal` to be consistent with `filtered_signal`, and since `get` usually implies no side-effects
 - Docstring
 - Change assertion to exception